index.js

```diff
@@ -1,192 +1,89 @@
-module.exports = transform
-
-var glslify = require('glslify-stream')
-  , deparser = require('glsl-deparser')
-  , replace = require('replace-method')
-  , concat = require('concat-stream')
-  , evaluate = require('static-eval')
-  , extract = require('glsl-extract')
-  , emit = require('emit-function')
-  , through = require('through')
-  , resolve = require('resolve')
-  , esprima = require('esprima')
-  , sleuth = require('sleuth')
-  , from = require('new-from')
-  , path = require('path')
-
-var usageRegex = /['"]glslify['"]/
-
-function transform(filename) {
-  var stream = through(write, end)
-    , accum = []
-    , len = 0
-
-  return stream
-
-  function write(buf) {
-    accum[accum.length] = buf
-    len += buf.length
-  }
+var glslifyBundle = require('glslify-bundle')
+var staticModule  = require('static-module')
+var glslifyDeps   = require('glslify-deps')
+var glslResolve   = require('glsl-resolve')
+var through       = require('through2')
+var nodeResolve   = require('resolve')
+var path          = require('path')
+var fs            = require('fs')
 
-  function end() {
-    var buf = Buffer.concat(accum).toString('utf8')
+module.exports = transform
 
-    // break out early if it doesn't look like
-    // we're going to find any shaders here,
-    // parsing and transforming the AST is expensive!
-    if(!usageRegex.test(buf)) {
-      return bail(buf)
+function transform(jsFilename) {
+  // static-module is responsible for replacing any
+  // calls to glslify in your JavaScript with a string
+  // of our choosing – in this case, our bundled glslify
+  // shader source.
+  var sm = staticModule({
+    glslify: bundle
+  }, {
+    vars: {
+      __dirname: path.dirname(jsFilename),
+      __filename: jsFilename,
+      require: {
+        resolve: nodeResolve
+      }
     }
+  })
 
-    var ast = esprima.parse(buf)
-      , name = glslifyname(ast)
-      , src = replace(ast)
-      , loading = 0
-      , map = {}
-      , id = 0
-
-    // bail early if glslify isn't required at all
-    if(!name) {
-      return bail(buf)
-    }
+  return sm
 
-    src.replace([name], function(node) {
-      var fragment
-        , current
-        , vertex
-        , config
+  function bundle(filename, opts) {
+    opts = opts || {}
 
-      current = ++id
+    var stream = through()
+    var depper = glslifyDeps({
+      readFile: readFile
+    })
 
-      if(!node.arguments.length) {
-        return
-      }
+    // Extract and add our local transforms.
+    var transforms = opts.transform || []
+    ;(Array.isArray(transforms)
+      ?  transforms
+      : [transforms]
```

Comment:
Ah, I meant that to mean: assign `var transforms = opts.transform || []; transforms = Array.isArray(transforms) ? transforms : [transforms];` instead of introducing a one-off array on line 43. This also lets us avoid the leading semicolon, and `transforms` would always point to an array.